### ISSUE TITLE
Javers diff is not working correctly for multi-dimensional arrays

### ISSUE THREAD

**Issue by cfranzen** [javers/javers#429]
Today I started to play around with Javers diff function and recognized that it is not working correctly for multi-dimensional arrays. I think it is a problem in ArrayChangeAppender. There the first dimension of the array is converted to a map and passed to MapChangeAppender which will do an Object.equals() on the second dimension arrays. Of course this will be not equal for two arrays, even if these arrays contain equal content. I prepared a test case ([JaversMultiDimArrayTest.zip](https://github.com/javers/javers/files/456795/JaversMultiDimArrayTest.zip)) to show that behavior. The output will look like:

```
Diff:
1. ArrayChange{globalId:'com.playground.JaversMultiDimArrayTest$TestEntity/', property:'intArray', containerChanges:[(0).'[I@4e885ea3'>>'[I@5afd479c', (1).'[I@2a68816f'>>'[I@3309b429']}
```

My expecatation would be that Javers traverses all dimensions of the array and compare the array values in the end.

**Comment by bartoszwalacik** [comment]
I would say that this is a missing feature not a problem or bug :)

**Comment by bartoszwalacik** [comment]
Agreed that JaVers should have some basic support for multi-dimensional arrays.
Second dimension could be compared with simple equals().

But fully-fledged diff would require more work (not something that we can do soon).

**Comment by suzzettedetorres** [comment]
Hello, any news on this topic?
I'm encountering the following error when comparing double[][]
java.lang.ClassCastException: [D cannot be cast to [Ljava.lang.Object;
at org.javers.core.metamodel.type.ArrayType.equals(ArrayType.java:96)

**Comment by bartoszwalacik** [comment]
if you need this feature, contribute :)

**Comment by bartoszwalacik** [comment]
closing due to lack of activity
